File: src/main/java/cube8540/oauth/authentication/SpringAsyncThreadConfiguration.java
package cube8540.oauth.authentication;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

import java.util.concurrent.Executor;

@EnableAsync
@Configuration
public class SpringAsyncThreadConfiguration {

    @Bean("asyncThreadPoolTaskExecutor")
    public Executor asyncThreadPoolTaskExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();

        executor.setCorePoolSize(4);
        executor.setMaxPoolSize(10);
        executor.setQueueCapacity(10);
        executor.setThreadNamePrefix("Spring-Async-");
        executor.initialize();

        return executor;
    }

}

File: src/main/kotlin/cube8540/oauth/authentication/SpringAsyncThreadConfiguration.kt
package cube8540.oauth.authentication

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import java.util.concurrent.Executor

@EnableAsync
@Configuration
class SpringAsyncThreadConfiguration {

    @Bean("asyncThreadPoolTaskExecutor")
    fun asyncThreadPoolTaskExecutor(): Executor {
        val executor = ThreadPoolTaskExecutor()

        executor.corePoolSize = 4
        executor.maxPoolSize = 10
        executor.setQueueCapacity(10)
        executor.setThreadNamePrefix("Spring-Async-")
        executor.initialize()

        return executor
    }
}